packages/@uppy/file-input/src/FileInput.tsx

```diff
@@ -1,39 +1,57 @@
-import { UIPlugin } from '@uppy/core'
+import { h, type ComponentChild } from 'preact'
+
+import { UIPlugin, Uppy, type UIPluginOptions } from '@uppy/core'
 import toArray from '@uppy/utils/lib/toArray'
-import { h } from 'preact'
+import type { Body, Meta } from '@uppy/utils/lib/UppyFile'
+import type { DefinePluginOpts } from '@uppy/core/lib/BasePlugin.js'
+import type { CSSProperties } from 'preact/compat'
 
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore We don't want TS to generate types for the package.json
 import packageJson from '../package.json'
-import locale from './locale.js'
+import locale from './locale.ts'
+
+export interface FileInputOptions<M extends Meta, B extends Body>
+  extends UIPluginOptions {
+  pretty?: boolean
+  inputName?: string
+}
+// Default options, must be kept in sync with @uppy/react/src/FileInput.js.
+const defaultOptions = {
+  pretty: true,
+  inputName: 'files[]',
+}
+
+type Opts<M extends Meta, B extends Body> = DefinePluginOpts<
+  FileInputOptions<M, B>,
+  keyof typeof defaultOptions
+>
 
-export default  class FileInput extends UIPlugin {
+export default class FileInput<M extends Meta, B extends Body> extends UIPlugin<
+  Opts<M, B>,
```

Comment:
```suggestion
  Opts,
```

Reply:
what's the point of these generics if we don't use them?